upgrade to net-8.0 and bump nugets to latest version

artwks opened the pull request:
This PR Contains:

1. .net 8 upgrade
2. updated all nugets to latest version

Tests are all passing:

<img width="584" alt="image" src="https://github.com/techops-recsys-lateral-hiring/developer-joyofenergy-csharp/assets/92155576/f4b7038d-e04a-48a0-bb57-524a9b04d3e9">

jejking-tw commented:
Please also update the README to reflect the latest state. (eg version of .NET).
Also - is there a dummy's guide to running this on MacOS?

artwks commented:
@jejking-tw Thank you for spotting this.

I have updated the readme with .net 8. I have also removed Visual Studio for Mac as it was dropped officially from Microsoft.

> is there a dummy's guide to running this on MacOS?
The existing steps in readme file covers the cli commands to run in macos.

Again Many thanks. Please let me know if I have missed anything else.
